src/1weso_test.cpp

```diff
@@ -37,13 +37,15 @@ int main() {
     fast_algorithm = false;
 
     uint64_t iter = 1000000;
-    OneWesolowskiCallback* weso = new OneWesolowskiCallback(D, iter);
-    FastStorage* fast_storage = NULL;
-    std::thread vdf_worker(repeated_square, f, D, L, weso, fast_storage, std::ref(stopped));
-    Proof proof = ProveOneWesolowski(iter, D, (OneWesolowskiCallback*)weso, stopped);
-    stopped = true;
-    vdf_worker.join();
-    free(weso);
+    Proof const proof = [&] {
+        OneWesolowskiCallback weso(D, iter);
+        FastStorage* fast_storage = nullptr;
+        std::thread vdf_worker(repeated_square, f, D, L, &weso, fast_storage, std::ref(stopped));
+        Proof proof = ProveOneWesolowski(iter, D, &weso, stopped);
+        stopped = true;
+        vdf_worker.join();
+        return proof;
+    }();
```

Comment:
is it going too far to use an immediately invoked lambda here? What it does it to allow the thread object and the `weso` object to destruct, while returning the `Proof` object, since we need that after this block comletes.

Reply:
Yeah, I also feel that the lambda here is a bit of overkill. Could you simply use `{}` for scoping here? IIRC the objects would also be destroyed when leaving the scope.